TestReport.fs

```
@@ -36,43 +37,56 @@ type Stdout = Stdout of (string * string) list
 type FailedTest = FailedTest of (ClassName * Name * Stdout * Backtrace option)
 
 let getFailedTests (xml:XmlDocument) =
-        let failed = 
+        let failed =
             xml.GetElementsByTagName("failure")
             |> Seq.cast<XmlNode>
 
-        let errors = 
+        let errors =
             xml.GetElementsByTagName("error")
             |> Seq.cast<XmlNode>
 
         Seq.append failed errors
             |> Seq.toArray
-            |> Array.Parallel.map (fun f -> 
+            |> Array.Parallel.map (fun f ->
                 let parent = f.ParentNode
                 let className =
                     parent
                         |> classname
                         |> ClassName
 
-                let n = 
+                let testName =
                     parent
                         |> name
                         |> Name
-        
+
                 let msg = message f
 
-                let stdout = 
+                let stdout =
                     run stdoutParser f.InnerText
                     |> getResult
                     |> Stdout
 
-                let x = parseTraceFromMessage msg
+                let trace = parseTraceFromMessage msg
 
-                FailedTest(className, n, stdout, x)
+                FailedTest(className, testName, stdout, trace)
             )
 
+let regex x =
+    Regex("lotus-\\d{1,3}vm\\d{1,2}").Replace(x, "<lotus-nodename>")
```

Wouldn't it be more useful to just apply a `s/lotus-\\d{1,3}//;s/vm\d+(\d)/vm\1/` to the hostname to preserve the _vmX_ name?  Names will then be reduced and normalized to their respective _vm[2-9]_ names rather than all hostnames being reduced down to a single generic string.

---

discussed this with @jgrund and decided we didn't want to mask failures that happened on different host combinations

---

> Wouldn't it be more useful to just apply a s/lotus-\\d{1,3}//;s/vm\d+(\d)/vm\1/ to the hostname to preserve the vmX name? Names will then be reduced and normalized to their respective vm[2-9] names rather than all hostnames being reduced down to a single generic string.

If we do that, then we will reduce the amount of aggregation possible though. 

This transform is only run when we aggregate failed jobs for reporting. All other times, the full string is preserved.

---

On the other hand, by conflating all of the hostnames to a single string, we could be falsely mis-matching results.

I have written a **lot** of text and log parsing code and I personally like start out as conservative as I can and see if/where I need to be more illiberal since it's usually easier to notice errors due to being too conservative rather than being too liberal.  The former typically stand out as errors and the latter get hidden as false positives.  Just my opinion/experience though.